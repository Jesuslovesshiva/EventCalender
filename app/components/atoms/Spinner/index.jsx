import classNames from 'classnames'
import PropTypes from 'prop-types'
import React from 'react'
import {FormattedMessage} from 'react-intl'

import commonMessages from '@/components/common-messages'

const SPINNER_TYPES = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary'
}

const SPINNER_TYPES_CLASSES = {
  [SPINNER_TYPES.PRIMARY]: 'border-primary/20 border-l-primary',
  [SPINNER_TYPES.SECONDARY]: 'border-gray-light border-l-gray-medium'
}

const Spinner = ({className, type}) => (
  <div
    className={classNames(
      'animate-spin',
      'rounded-full w-8 h-8',
      'border-4',
      SPINNER_TYPES_CLASSES[type],
      className
    )}
    role="status"
  >
    <span className="sr-only">
      <FormattedMessage {...commonMessages.loading} />
    </span>
  </div>
)

Spinner.propTypes = {
  className: PropTypes.any,
  type: PropTypes.oneOf(Object.values(SPINNER_TYPES))
}

Spinner.defaultProps = {
  className: null,
  type: SPINNER_TYPES.PRIMARY
}

export {SPINNER_TYPES}
export default React.memo(Spinner)
