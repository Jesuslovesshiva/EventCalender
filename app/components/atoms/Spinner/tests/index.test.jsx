import commonMessages from '@/components/common-messages'
import render from '@/utils/test-utils/custom-render'

import Spinner, {SPINNER_TYPES} from '../index'

const renderComponent = (props = {}, renderFn = render) =>
  renderFn(<Spinner {...props} />)

describe('Spinner', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a screen reader only text', () => {
    const {getByText} = renderComponent()
    expect(getByText(commonMessages.loading.defaultMessage)).toBeInTheDocument()
  })

  it('should render the correct class names', () => {
    const {container, rerender} = renderComponent()
    expect(container.firstChild).toHaveClass('border-l-primary')
    renderComponent({type: SPINNER_TYPES.SECONDARY}, rerender)
    expect(container.firstChild).not.toHaveClass('border-l-primary')
    expect(container.firstChild).toHaveClass('border-l-gray-medium')
  })
})
