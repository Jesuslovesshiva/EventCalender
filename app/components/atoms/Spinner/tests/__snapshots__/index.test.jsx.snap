// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should match the snapshot 1`] = `
<div
  class="animate-spin rounded-full w-8 h-8 border-4 border-primary/20 border-l-primary"
  role="status"
>
  <span
    class="sr-only"
  >
    Wird geladen...
  </span>
</div>
`;
